internal/bminventory/inventory.go

```diff
@@ -4048,6 +4054,19 @@ func (b *bareMetalInventory) validateClusterInfraEnvRegister(ctx context.Context
 	return nil
 }
 
+func (b *bareMetalInventory) setIgnitionConfigOverrideUsage(clusterId strfmt.UUID, ignitionConfigOverride string, db *gorm.DB) error {
+	var cluster *common.Cluster
+	cluster, err := common.GetClusterFromDB(db, clusterId, common.SkipEagerLoading)
```

Comment:
We'll need FOR UPDATE here as well

Reply:
Thanks for catching this! I switched to the correct function.